src/OrbitQt/HistogramWidget.h

```diff
@@ -19,15 +24,33 @@ class HistogramWidget : public QWidget {
  public:
   using QWidget::QWidget;
 
-  void UpdateHistogram(std::optional<orbit_statistics::Histogram> histogram,
-                       std::string function_name);
+  void UpdateData(std::vector<uint64_t> data, std::string function_name);
 
  protected:
   void paintEvent(QPaintEvent* /*event*/) override;
 
+  void mousePressEvent(QMouseEvent* event) override;
+  void mouseReleaseEvent(QMouseEvent* event) override;
+  void mouseMoveEvent(QMouseEvent* event) override;
+
  private:
-  std::optional<orbit_statistics::Histogram> histogram_;
-  std::optional<std::string> function_name_;
+  [[nodiscard]] bool IsSelectionActive() const;
+
+  [[nodiscard]] uint64_t MinValue() const;
+  [[nodiscard]] uint64_t MaxValue() const;
+
+  [[nodiscard]] int Width() const;
+  [[nodiscard]] int Height() const;
+  [[nodiscard]] int WidthMargin() const;
+  [[nodiscard]] int HeightMargin() const;
+
+  std::optional<const std::vector<uint64_t>> data_;
+  std::optional<const std::string> function_name_;
+
+  std::stack<orbit_statistics::Histogram> histogram_stack_;
+
+  std::optional<int> selection_start_pixel;
+  std::optional<int> selection_current_pixel;
```

Comment:
It looks to me if these two members share their lifetime. What about putting them into a struct and putting the struct into an optional?

Reply:
Also the variable names should end with an underscore, i.e. `selection_start_pixel_`.